### ISSUE TITLE
Security: API allows for anonymous link creation

### ISSUE THREAD

**Issue by dewittn** [thedevs-network/kutt#553]
I have disabled anonymous link creation with the fallowing setting `DISALLOW_ANONYMOUS_LINKS=true` but I am still able to create links via the API **without** providing an API KEY. 

```
curl --header "Content-Type: application/json" \
--request POST \
--data '{"target": "https://google.com"}' \
http://127.0.0.1:3000/api/url/submit
```

This should **not** work but I get back a shortened link. Worse yet, there is no place in the UI that displays informatin about the anonymous links that have been generated. This allows anyone with the URL of a Kutt instillation to create links without the admin's knowledge.

Would be great if someone else could confirm this. 

I'm running Kutt 2.7.3 on Ubuntu 20.04.3 LTS with node 12.22.9.

**Comment by elimisteve** [comment]
@dewittn The cause of the issue you describe above could be the same underlying cause that I've been experiencing in several cases: all customizations I make to the configuration are being ignored.

Is whatever you set for `SITE_NAME` visible at `/terms`? For me this is yet another environment variable not being set (or not being respected).

Also, when I set `PORT` to anything other than `3000`, it doesn't work. I think the port is hardcoded, as are the Postgres username and password, even though `.docker.env` encourages us to set those variables! https://github.com/thedevs-network/kutt/blob/develop/docker-compose.yml

**Comment by elimisteve** [comment]
The custom environment vars I'm setting in `.env` (copied from `.docker.env` as per [the docs](https://github.com/thedevs-network/kutt/#docker)) don't seem to make a difference even when they're not being ignored in the code like they are in the `docker-compose.yml` example mentioned at the end of my previous comment.

This seems to be because

```js
import getConfig from "next/config"

...

const { publicRuntimeConfig } = getConfig();

...

// Anything involving publicRuntimeConfig.SITE_NAME or publicRuntimeConfig.DEFAULT_DOMAIN or publicRuntimeConfig.SOME_ENV_VAR_NAME_HERE
```

results in variables like `publicRuntimeConfig.SITE_NAME` being `undefined`. Any idea if Next.js needs to be hosting the site? I did an `npm run build` and assumed that I could simply put the resulting build behind Nginx and have everything work, but maybe we need to do something special to get Next.js's `getConfig` function to parse the contents of `.env`?

In `.env` I tried adding `export ` to the beginning of `SITE_NAME=...`, but that got me a syntax error.

I don't know what the solution is yet, but I'll reply in this thread if I figure it out.

**Comment by dewittn** [comment]
@elimisteve I have Kutt running behind nginx and it works just fine, so I don't think that's the problem.

In my testing, some of the variables have an effect on the site. For example, I can turn registration on and off without any issues. But for whatever reason changing the `DISALLOW_ANONYMOUS_LINKS` variable has no effect.

 You do have to completely restart the app for the changes to take.

**Comment by elimisteve** [comment]
@dewittn That's helpful! Is your Kutt running inside Docker?

**Comment by dewittn** [comment]
@elimisteve No I have mine installed "manually" but theoretically there shouldn't be much difference. I'm using Nginx as a reverse proxy to handle SSL and requests from multiple domains. Nginx then passes it off to the node server running on port `3000`. 

I had to set `CUSTOM_DOMAIN_USE_HTTPS` to `true` but that's about it.

**Comment by elimisteve** [comment]
> No I have mine installed "manually" but theoretically there shouldn't be much difference.

Indeed, but when the backend reloads `.env` is probably one of the differences, and it's good to know that all you had to do was restart the backend to get those updated env vars to be respected (when they are at all).

FYI I'm running Kutt using Docker Compose, but the other parts of the setup are the same as yours: Nginx is in front, terminating SSL then proxying through to port 3000 where Kutt is running.

Thanks again!

**Comment by poeti8** [comment]
@thedevs-network/theguard  Thanks for the report. I fixed the issue on 2.7.4. The problem was I forgot to consider and apply `DISALLOW_ANONYMOUS_LINKS` on v1 (old API) as well.

@elimisteve The problem you mentioned however, is different.  Please check https://github.com/thedevs-network/kutt/issues/388 for more information, I have yet to fix this.

**Comment by poeti8** [comment]
@dewittn Thanks for the report. I fixed the issue on 2.7.4. The problem was I forgot to consider and apply `DISALLOW_ANONYMOUS_LINKS` on v1 (old API) as well.

@elimisteve The problem you mentioned however, is different.  Please check https://github.com/thedevs-network/kutt/issues/388 for more information, I have yet to fix this.
